mlflow/utils/_spark_utils.py

```diff
@@ -32,6 +32,7 @@ def _create_local_spark_session_for_pipelines():
         .config(
             "spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog"
         )
+        .config("spark.sql.execution.arrow.pyspark.enabled", "true")
```

Comment:
Is this change intentional?

Reply:
Yes, this change is necessary. Why? The predict step reads data from the ingest step by first reading it with pandas and then converting the pandas DF into a spark DF. We do not directly read with spark because spark cannot read directly from the driver disk. However, in this conversion process, without "arrow" being enabled, the spark DF will change the datatypes of the pandas DF in a way that breaks the input schema of the registered model. Hence, I added this configuration to spark.